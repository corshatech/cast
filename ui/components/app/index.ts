export * from "./Layout/Layout";
export * from "./FeatureComponent/FeatureComponent";
