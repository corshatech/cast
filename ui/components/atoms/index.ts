export * from './Button/Button';
export * from './Typography/Typography';
