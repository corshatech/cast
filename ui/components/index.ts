export * from './atoms';
export * from './app';
