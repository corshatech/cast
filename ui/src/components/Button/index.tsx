import React from "react";
import css from "./Button.module.scss";

interface ButtonProps {
  /**
   * Is this the principal call to action on the page?
   */
  primary?: boolean;
  /**
   * What background color to use
   */
  backgroundColor?: string;
  /**
   * How large should the button be?
   */
  size?: "small" | "medium" | "large";
  /**
   * Button contents
   */
  label: string;
  /**
   * Optional click handler
   */
  onClick?: () => void;
}

/**
 * Primary UI component for user interaction
 */
export const Button = ({
  primary = false,
  size = "medium",
  backgroundColor,
  label,
  ...props
}: ButtonProps) => {
  const mode = primary
    ? css["storybook-button-primary"]
    : css["storybook-button-secondary"];
  return (
    <button
      type="button"
      className={[
        css["storybook-button"],
        css[`storybook-button-${size}`],
        mode,
      ].join(" ")}
      style={{ backgroundColor }}
      {...props}
    >
      {label}
    </button>
  );
};
