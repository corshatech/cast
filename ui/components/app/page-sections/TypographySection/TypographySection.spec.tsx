import renderer from 'react-test-renderer';
import { TypographySection } from '@/components/index';

describe('Typography Section', () => {
  it('snapshot', () => {
    const tree = renderer.create(<TypographySection />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
