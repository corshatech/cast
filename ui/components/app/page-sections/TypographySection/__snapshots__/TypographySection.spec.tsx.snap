// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Typography Section snapshot 1`] = `
[
  <div
    className="mb-8 mt-24"
  >
    <h2
      className="leading-6 text-h2 font-poppins font-bold"
    >
      Typography
    </h2>
  </div>,
  <hr
    className="my-12"
  />,
  <div
    className="space-y-6 mb-24 flex flex-col"
  >
    <h1
      className="leading-6 text-h1 font-poppins font-bold"
    >
      Heading Level One
    </h1>
    <h2
      className="leading-6 text-h2 font-poppins font-bold"
    >
      Heading Level Two
    </h2>
    <h3
      className="leading-6 text-h3 font-poppins font-bold"
    >
      Heading Level Three
    </h3>
    <h4
      className="leading-6 text-h4 font-poppins font-semibold"
    >
      Heaving Level Four
    </h4>
    <div
      className="grid grid-cols-4 justify-between items-start pt-12"
    >
      <p
        className="leading-6 text-body1 font-nunito"
      >
        <span
          className="font-black"
        >
          Body 1 Black
        </span>
        <span
          className="font-black"
        >
          Nunito Sans Black 16px
        </span>
      </p>
      <p
        className="leading-6 text-body1 font-nunito"
      >
        <span
          className="font-bold"
        >
          Body 1 Bold
        </span>
        <span
          className="font-bold"
        >
          Nunito Sans Bold 16px
        </span>
      </p>
      <p
        className="leading-6 text-body1 font-nunito"
      >
        <span>
          Body 1 Regular
        </span>
        <span>
          Nunito Sans Regular 16px
        </span>
      </p>
      <p
        className="leading-6 text-body1 font-nunito"
      >
        <span
          className="underline font-medium"
        >
          Body 1 Underlined
        </span>
        <span
          className="underline font-medium"
        >
          Nunito Sans Medium Underline 16px
        </span>
      </p>
    </div>
    <div
      className="grid grid-cols-4 justify-between items-start pt-12"
    >
      <p
        className="leading-6 text-body2 font-nunito"
      >
        <span
          className="font-bold"
        >
          Body 2 Bold
        </span>
        <span
          className="font-bold"
        >
          Nunito Sans Bold 14px
        </span>
      </p>
      <p
        className="leading-6 text-body2 font-nunito"
      >
        <span>
          Body 2 Regular
        </span>
        <span>
          Nunito Sans Regular 14px
        </span>
      </p>
      <p
        className="leading-6 text-body2 font-nunito"
      >
        <span
          className="underline font-medium"
        >
          Body 2 Underlined
        </span>
        <span
          className="underline font-medium"
        >
          Nunito Sans Medium Underline 14px
        </span>
      </p>
      <p
        className="leading-6 text-body2 font-nunito"
      >
        <span
          className="font-medium"
        >
          All Caps
        </span>
        <span
          className="font-black capitalize"
        >
          Nunito Sans Black All Caps 14px
        </span>
      </p>
    </div>
    <div
      className="grid grid-cols-4 justify-between items-start pt-12"
    >
      <p
        className="leading-6 text-body3 font-nunito"
      >
        <span
          className="font-black"
        >
          Body 3 Black
        </span>
        <span
          className="font-black"
        >
          Nunito Sans Black 12px
        </span>
      </p>
      <p
        className="leading-6 text-body3 font-nunito"
      >
        <span
          className="font-bold"
        >
          Body 3 Bold
        </span>
        <span
          className="font-bold"
        >
          Nunito Sans Bold 12px
        </span>
      </p>
      <p
        className="leading-6 text-body3 font-nunito"
      >
        <span>
          Body 3 Regular
        </span>
        <span>
          Nunito Sans Regular 12px
        </span>
      </p>
      <p
        className="leading-6 text-body3 font-nunito"
      >
        <span
          className="underline font-medium"
        >
          Body 3 Underlined
        </span>
        <span
          className="underline font-medium"
        >
          Nunito Sans Medium Underline 12px
        </span>
      </p>
    </div>
    <div
      className="grid grid-cols-4 justify-between items-start pt-12"
    >
      <p
        className="leading-6 text-body4 font-nunito"
      >
        <span
          className="font-black"
        >
          Body 4 Black
        </span>
        <span
          className="font-black"
        >
          Nunito Sans Black 10px
        </span>
      </p>
      <p
        className="leading-6 text-body4 font-nunito"
      >
        <span
          className="font-bold"
        >
          Body 4 Bold
        </span>
        <span
          className="font-bold"
        >
          Nunito Sans Bold 10px
        </span>
      </p>
      <p
        className="leading-6 text-body4 font-nunito"
      >
        <span>
          Body 4 Regular
        </span>
        <span>
          Nunito Sans Regular 10px
        </span>
      </p>
      <p
        className="leading-6 text-body4 font-nunito"
      >
        <span
          className="underline font-medium"
        >
          Body 4 Underlined
        </span>
        <span
          className="underline font-medium"
        >
          Nunito Sans Medium Underline 10px
        </span>
      </p>
    </div>
  </div>,
  <div
    className="h-64"
  />,
]
`;
